dataset/src/main/scala/frameless/RecordEncoder.scala

```diff
@@ -162,23 +161,13 @@ class RecordEncoder[F, G <: HList, H <: HList]
 
     def fromCatalyst(path: Expression): Expression = {
       val exprs = fields.value.value.map { field =>
-        val fieldPath = path match {
-          case BoundReference(ordinal, dataType, nullable) =>
```

Comment:
@chris-twiner was this part of the code causing issues? I am trying to think if this has any unexpected side effects. I am already seeing some difference with the previous version here:

In frameless 0.8
```
scala> val x = TypedDataset.create(Array(1,2,3,1))
x: frameless.TypedDataset[Int] = [_1: int]
```

In new PR 
```
scala> val x = TypedDataset.create(Array(1,2,3,1))
x: frameless.TypedDataset[Int] = [value: int]
```

At least the field name seems to be printed differently (was `_1` before but not it's `value`).

Reply:
The problem is it's not set by frameless code, rather by the [expression encoder](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala#L270).  You can't inject the schema any more.  

But you are right its probably breaking - I've updated the readme to mention this.